src/pages/groupComparison/pathwayMapper/GroupComparisonPathwayMapperUserSelectionStore.ts

```diff
@@ -0,0 +1,40 @@
+import { action, makeObservable, observable } from 'mobx';
+export interface IRankingChoices {
+    dropDownTitle: string;
+    isPercentageMatch: number;
+    isAlterationEnabled: number;
+    considerOnlyTCGAPanPathways: boolean;
+}
+
+export default class GroupComparisonPathwayMapperUserSelectionStore {
+    constructor() {
+        makeObservable(this);
+    }
+    @observable
+    currentPathway: string = '';
+    @observable
+    rankingChoices: IRankingChoices = {
+        dropDownTitle: 'Match count',
+        isPercentageMatch: 0,
+        isAlterationEnabled: 0,
+        considerOnlyTCGAPanPathways: true,
+    };
+    @observable selectedGenes: string[] | undefined = undefined;
+
+    @action.bound
+    updateRankingChoices(
+        dropDownTitle: string,
+        isAlterationEnabled: number,
+        considerOnlyTCGAPanPathways: boolean,
+        isPercentageMatch: number,
+        selectedPathway: string
+    ) {
+        this.currentPathway = selectedPathway;
+        Object.assign(this.rankingChoices, {
```

Comment:
if you are always overwriting all the properties, you might as well overwrite the whole object and make the object @obserable.ref, which is a more performance form of observability.  However, it's negligible improvement here.